### PR TITLE
3.13t

### PR DESCRIPTION
**Pull request by ChrisLovering**
Add support for running nsjail under 3.13t (free-threaded interpreter)

### CODE REVIEW

File: scripts/build_python.sh [hunk 1]
```diff
@@ -4,11 +4,19 @@ shopt -s inherit_errexit
 
 py_version="${1}"
 
-# Install Python interpreter under e.g. /snekbin/python/3.11/ (no patch version).
+# Install Python interpreter under e.g. /snekbin/python/3.13/ (no patch version)
+# By dropping everything after, and including, the last period.
+install_path="${1%[-.]*}"
```
Comment (MarkKoz) [inline scripts/build_python.sh:9]: Nit: use `py_version` instead of `1`

File: tests/test_integration.py [hunk 2]
```diff
@@ -83,6 +83,25 @@ def test_alternate_executable_support(self):
...
+            }
+            cases = [
+                ("3.13", "0\n"),
+                ("3.13t", "1\n"),
+            ]
+            for version, expected in cases:
+                with self.subTest(version=version, expected=expected):
+                    response, status = snekbox_request(
+                        get_gil_status
+                        | {"executable_path": f"/snekbin/python/{version}/bin/python"}
```
Comment (MarkKoz) [inline tests/test_integration.py:100]: Nit: can we make this more readable by creating the dict on a separate line and saving it in a variable?

File: requirements/eval-deps.pip [hunk 3]
```diff
@@ -7,7 +7,7 @@ fishhook~=0.3; python_version == '3.12'
 forbiddenfruit~=0.1
 fuzzywuzzy~=0.18
 lark~=1.2
-matplotlib~=3.9
+matplotlib~=3.9 ; python_version != '3.13'
```
Comment (MarkKoz) [inline requirements/eval-deps.pip:10]: I take it there's (currently) no way to only exclude nogil builds?

**Review by MarkKoz — APPROVED** [review]

File: requirements/eval-deps.pip (hunk 3 above)
Comment (ChrisLovering) [reply requirements/eval-deps.pip:10]: unfortunatly not right now.
https://discuss.python.org/t/environment-marker-for-free-threading/60007/5
I'll force push the commit to link to this discourse thread

File: scripts/build_python.sh (hunk 1 above)
Comment (ChrisLovering) [reply scripts/build_python.sh]: https://github.com/python-discord/snekbox/pull/218/commits/73e12c84ecb8e3ece9a266f40cee96264efee324

File: tests/test_integration.py (hunk 2 above)
Comment (ChrisLovering) [reply tests/test_integration.py]: https://github.com/python-discord/snekbox/pull/218/commits/7551599a8a085d27b48a233032fe58fcd738cb80

**Pull request merged**